canonical-web-and-design/ubuntu.com
# Remove architecture selection logic from /core/build

## Done

- Let the user choose their target architecture instead of it being chosen for them based on their board and OS
- Reveal all Snaps, regardless of what options the user has previously entered

## QA

- Check out this feature branch
- In `webapp/views.py`, find the `post_build()` function. Below the line that reads :
```python
arch = flask.request.values.get("arch")
```
paste the following code: 
```python
    if opt_in:
        print("Opt-in: " + opt_in)
    print("Full name: " + full_name)
    print("Email: " + email)
    print("Board: " + board)
    print("System: " + system)
    print("Architecture: " + arch)
    print("Snaps:")
    for snap in snaps:
        print("\t" + snap)

    return flask.render_template("core/build/index.html")
```
and then comment out the rest of the function.
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/core/build
- Sign in with Ubuntu One, fill out the form and press `Build image`.
  - Ensure that each field only becomes enabled when the previous one is filled out
  - Ensure that all your selected options printed correctly


## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/3090

## NOTE:
This PR is dependent on [this one](https://github.com/canonical-web-and-design/canonicalwebteam.launchpad/pull/32), which allows the Launchpad.build_image function to optionally accept the architecture argument that was added here. Do not merge until both are approved.
